tools/pihole_toml_to_markdown.py

```diff
@@ -0,0 +1,229 @@
+
+            # Process the comments collected for this key
+            if comment_buffer:
+                adjusted_comments = []
+
+                i = 0
+                while i < len(comment_buffer):
+                    line = comment_buffer[i]
+                    is_bullet = line.lstrip().startswith("-")
+                    prev_is_bullet = i > 0 and comment_buffer[i - 1].lstrip().startswith("-")
+                    prev_is_blank = i > 0 and comment_buffer[i - 1].strip() == ""
+
+                    # Fix malformed emphasis due to underscores or asterisks
+                    line = re.sub(r'\b(_[a-zA-Z0-9.-]+)', r'`\1`', line)
+                    line = re.sub(r'\*\.[a-zA-Z0-9]+', lambda m: f"`{m.group(0)}`", line)
+
+                    # Bold "Allowed values are:"
+                    line = re.sub(
+                        r'(^|\s)(Allowed values are:)', 
+                        r'\1**Allowed values are:**', 
+                        line
+                    )                   
+
+                    # Insert blank line before bullet if needed
+                    if is_bullet and not prev_is_bullet and not prev_is_blank:
+                        adjusted_comments.append("")
+
+                    # Handle Example/Example n: lines that may be split across multiple lines
+                    example_match = re.match(r"^(Example(?: \d*)?:)(.*)", line)
+                    if example_match:
+                        example_label = example_match.group(1)
+                        example_content = example_match.group(2).strip()
+                        # Collect continuation lines (those that are indented or start with # or ")
+                        j = i + 1
+                        while j < len(comment_buffer):
+                            next_line = comment_buffer[j]
+                            if next_line.startswith(" ") or next_line.startswith('"') or next_line.startswith("[") or next_line.startswith("#"):
+                                example_content += " " + next_line.strip()
+                                j += 1
+                            else:
+                                break
+                        adjusted_comments.append(f"**{example_label}** `{example_content}`")
+                        i = j  # Skip the lines we've just consumed
+                        continue                    
+                   
+                    # Default: just append the line
+                    adjusted_comments.append(line)
+                    i += 1
+
+                documentation.append(escape_angle_brackets_outside_backticks("\n".join(adjusted_comments)))
+                documentation.append("")  # spacer after comment block
+            # Format default value for Markdown
+            if "\n" in value:
+                documentation.append("**Default value:**")
+                documentation.append("```toml")
+                documentation.append(f"{value}")
+                documentation.append("```")
```

Comment:
```suggestion
            if "\n" in value:
                documentation.append("**Default value:**")
                documentation.append("")
                documentation.append("```toml")
                documentation.append(f"{value}")
                documentation.append("```")
                documentation.append("")
```

Reply:
Changed locally, will be reflected in the next FP